src/py/pyodide/console.py

```diff
@@ -488,10 +488,40 @@ def repr_shorten(
     If it is longer than ``limit`` then return the firsts ``split``
     characters and the last ``split`` characters separated by '...'.
     Default value for ``split`` is `limit // 2`.
+    If ``split`` is longer than the length of ``len(value) // 2``, ``split`` will be `len(value) / 2`.
+
+    Examples
+    --------
+    >>> from pyodide.console import repr_shorten
+    >>> sep = "_"
+    >>> for i in range(6):
+    ...     repr_shorten("1234", limit=i, separator=sep)
+    "_'1234'"
+    "_'1234'"
+    "'_'"
+    "'_'"
+    "'1_4'"
+    "'1234'"
+    >>> for i in range(6):
+    ...     repr_shorten("1234", limit=i, split=i, separator=sep)
+    "_'1234'"
+    "'_'"
+    "'1_4'"
+    "'12_34'"
+    "'12_34'"
+    "'1234'"
+    >>> len("0123456789")
+    10
+    >>> repr_shorten("0123456789", limit=1, split=4, separator=sep)
+    "'012_789'"
+    >>> repr_shorten("0123456789", limit=15, split=4, separator=sep)
+    "'0123456789'"
     """
     if split is None:
         split = limit // 2
     text = repr(value)
-    if len(text) > limit:
+    if split > len(text) / 2:
+        split = int(len(text) / 2)
+    if len(text) - 2 > limit:
```

Comment:
> If it is longer than ``limit`` then return the firsts ``split``
> characters and the last ``split`` characters separated.

The correct repr_shorten output as I understand it is:
```python
>>> for i in range(10):
...     repr_shorten("12345678", limit=i, split=i, separator=sep)

"linit:0 | split:0 | _'12345678'"
"linit:1 | split:1 | '_'"
"linit:2 | split:2 | '1_8'"
"linit:3 | split:3 | '12_78'"
"linit:4 | split:4 | '123_678'"
"linit:5 | split:5 | '1234_5678'"
"linit:6 | split:5 | '1234_5678'"
"linit:7 | split:5 | '1234_5678'"
"linit:8 | split:5 | '12345678'"
"linit:9 | split:5 | '12345678'"
```

So I thought I needed to fix two things.
1. Shouldn't this be fixed because using `repr` includes `''` and adds 2 to the length of the string? `if len(text) - 2 > limit:`
2. Shouldn't `split' just compare if it's less than half the length of the original string and change the value if it's greater?

Can you please confirm what I'm misunderstanding?

Reply:
1. I think it is supposed to apply repr and then shorten the result. Applying repr to a string adds quotes. It would probably be a better API to require `value` to be a `str` and only do the shortening and then call it like `repr_shorten(repr(v))`. Currently that would be an API break. Maybe add a separate `shorten` function and redefine `repr_shorten` to do `shorten(repr(v), other args)`. That way if you already have a string that you want to shorten it won't gain quotes.
2. Yeah something like that sounds good.

Also, I think if limit is very small we should throw a value error (does it make any sense for limit to be 0 or 1?)
Also, regardless of what we do to limit the value of `split` we should make sure we don't see cases like you show where the separator is added but no characters are removed. Maybe something like:
```py
if len(shortened_string) >= len(original_string):
   return original_string
```
would be good.